### PR TITLE
feat: replace django-saml2-auth with djangosaml2

### PR DESCRIPTION
**Pull request by alles-klar**
Issue: https://github.com/DefectDojo/django-DefectDojo/issues/3890

Documentation is part of the commit.

----

On behalf of DB Systel GmbH

### CODE REVIEW

**Comment by devGregA** [comment]
@alles-klar  haven't forgotten about this, feedback coming shortly.

File: dojo/settings/settings.dist.py [hunk 1]
```diff
@@ -703,6 +672,148 @@ def generate_url(scheme, double_slashes, user, password, host, port, path, param
...
 
+# ------------------------------------------------------------------------------
+# SAML
+# ------------------------------------------------------------------------------
+# For more configuration and customization options, see djangosaml2 documentation
+# https://djangosaml2.readthedocs.io/contents/setup.html#configuration
+# To override not configurable settings, you can use local_settings.py
+SAML2_ENABLED = env('DD_SAML2_ENABLED')
+SAML2_LOGOUT_URL = env('DD_SAML2_LOGOUT_URL')
+if env('DD_SAML2_ENABLED'):
```
Comment (Maffooch) [inline dojo/settings/settings.dist.py:683]: Probably best to `SAML2_ENABLED` to accommodate users not using env vars. Also the env var is already stored there

**Review by Maffooch — COMMENTED** [review]
This package is a little more complicated than the old one, but way more configurable. Definitely a good move from a maintainability perspective.

I migrated my dev instance from the old to the new, and it wasn't too painful. Lots of reading the docs and debugging to get the gist of things, but that's kinda unavoidable with anything saml related.

Will approve on comment resolution.

File: dojo/settings/settings.dist.py [hunk 2]
```diff
@@ -120,7 +120,7 @@
     DD_SAML2_ATTRIBUTES_MAP=(dict, {
         # Change Email/UserName/FirstName/LastName to corresponding SAML2 userprofile attributes.
         'Email': ('email', ),
-        # 'UserName': ('username', ),
+        'UserName': ('username', ),
```
Comment (alles-klar) [inline dojo/settings/settings.dist.py:123]: @Maffooch Does this work in your setup? In the past I had problems when mapping username here and also setting SAML_DJANGO_USER_MAIN_ATTRIBUTE = 'username'.

Comment (Maffooch) [reply dojo/settings/settings.dist.py:123]: My org does not have a username attribute to provide, so I put the email attribute in both the email and username field on the user
```
SAML_ATTRIBUTE_MAPPING =  {
        # Change Email/UserName/FirstName/LastName to corresponding SAML2 userprofile attributes.
        'email': ('email', 'username'),
        'firstName': ('first_name', ),
        'lastName': ('last_name', ),
    }
```

Comment (alles-klar) [reply dojo/settings/settings.dist.py:123]: well, users have to adopt there configuration anyway

**Review by valentijnscholten — APPROVED** [review]

**Review by Maffooch — APPROVED** [review]

**Pull request merged**